include/irr/asset/filters/CSummedAreaTableImageFilter.h

```diff
@@ -0,0 +1,382 @@
+				{
+					inOffsetBaseLayer = other.inOffsetBaseLayer;
+					inExtentLayerCount = other.inExtentLayerCount;
+					outOffsetBaseLayer = other.outOffsetBaseLayer;
+					outExtentLayerCount = other.outExtentLayerCount;
+				}
+				virtual ~CPrototypeState() {}
+
+				union
+				{
+					core::vectorSIMDu32	inOffsetBaseLayer;
+					struct
+					{
+						VkOffset3D		inOffset;
+						uint32_t		inBaseLayer;
+					};
+				};
+				union
+				{
+					core::vectorSIMDu32 inExtentLayerCount;
+					struct
+					{
+						VkExtent3D		inExtent;
+						uint32_t		inLayerCount;
+					};
+				};
+				union
+				{
+					core::vectorSIMDu32 outOffsetBaseLayer;
+					struct
+					{
+						VkOffset3D		outOffset;
+						uint32_t		outBaseLayer;
+					};
+				};
+				union
+				{
+					core::vectorSIMDu32 outExtentLayerCount;
+					struct
+					{
+						VkExtent3D		outExtent;
+						uint32_t		outLayerCount;
+					};
+				};
+
+				uint32_t				inMipLevel = 0u;
+				uint32_t				outMipLevel = 0u;
+				ICPUImage* inImage = nullptr;
+				ICPUImage* outImage = nullptr;
+		};
+
+		class CState : public CPrototypeState, public CSummedAreaTableImageFilterBase<ExclusiveMode> {};
+		using state_type = CState; //!< full combined state
+
+		static inline bool validate(state_type* state)
+		{
+			if (!CSummedAreaTableImageFilterBase<ExclusiveMode>::validate(state))
+				return false;
+			
+			const ICPUImage::SCreationParams& inParams = state->inImage->getCreationParameters();
+			const ICPUImage::SCreationParams& outParams = state->outImage->getCreationParameters();
+			const auto inFormat = inParams.format;
+			const auto outFormat = outParams.format;
+
+			if (state->scratchMemoryByteSize < CSummedAreaTableImageFilterBase<ExclusiveMode>::getRequiredScratchByteSize(outFormat, {outParams.extent.width, outParams.extent.height, outParams.extent.depth}))
+				return false;
+
+			if (state->inLayerCount != state->outLayerCount)
+				return false;
+
+			IImage::SSubresourceLayers subresource = { static_cast<IImage::E_ASPECT_FLAGS>(0u),state->inMipLevel,state->inBaseLayer,state->inLayerCount };
+			if (!CBasicImageFilterCommon::validateSubresourceAndRange(subresource, { state->inOffset,state->inExtent }, state->inImage))
+				return false;
+
+			subresource.mipLevel = state->outMipLevel;
+			subresource.baseArrayLayer = state->outBaseLayer;
+			if (!CBasicImageFilterCommon::validateSubresourceAndRange(subresource, { state->outOffset,state->outExtent }, state->outImage))
+				return false;
+
+			// TODO: remove this later when we can actually write/encode to block formats
+			if (asset::isBlockCompressionFormat(outFormat))
+				return false;
+
+			if (asset::getFormatChannelCount(outFormat) < asset::getFormatChannelCount(inFormat))
+				return false;
+
+			if (getFormatClass(state->inImage->getCreationParameters().format) <= getFormatClass(outFormat))
+				return false;
```

Comment:
if `CSummedAreaTableImageFilterBase<ExclusiveMode>` inherited from matched extent base then you could call its validation and have all this already be taken care of

Reply:
adjusted